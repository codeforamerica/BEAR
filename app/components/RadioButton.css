.radioButtonWrapper {
  display: inline-block !important;
  padding: 0 !important;
  height: 54px;
  width: 54px;
  border: 2px solid black;
}

.radioButtonWrapper:focus-within {
  box-shadow: none !important;
}

.selected {
  background-color: #ebfffa !important;
}

.radioButton {
  -webkit-appearance: none;
  top: 14px;
  left: 14px;
  height: 22px;
  width: 22px;
  padding: 0.3em !important;
  border: 2px solid #30302f !important;
  border-radius: 50% !important;
}

.radioButton:checked::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -6px;
  margin-left: -6px;
  height: 12px;
  width: 12px;
  border-radius: 50%;
  background-color: #017f6a;
}
