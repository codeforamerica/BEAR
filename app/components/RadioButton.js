import React, { Component } from 'react';
import styles from './RadioButton.css';

type Props = {
  selected: boolean,
  value: string,
  group: string
};

export default class RadioButton extends Component<Props> {
  render() {
    const { selected, value, group } = this.props;

    return (
      <input
        defaultChecked={selected}
        type="radio"
        className={styles.radioButton}
        name={group}
        id={`${value}_${group}`}
      />
    );
  }
}
