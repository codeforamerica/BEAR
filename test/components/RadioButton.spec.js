import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import RadioButton from '../../app/components/RadioButton';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.sandbox.create();

function setup(selected) {
  const component = shallow(
    <RadioButton selected={selected} value="whatever" group="nihilists" />
  );
  return {
    component
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('RadioButton component', () => {
  it('is checked when selected is true', () => {
    const { component } = setup(true);
    expect(component.find('input').props().defaultChecked).toEqual(true);
  });

  it('is not checked when selected is false', () => {
    const { component } = setup(false);
    expect(component.find('input').props().defaultChecked).toEqual(false);
  });

  it('sets the name to the group prop', () => {
    const { component } = setup(false);
    expect(component.find('input').props().name).toEqual('nihilists');
  });

  it('constructs the id from the group and value', () => {
    const { component } = setup(false);
    expect(component.find('input').props().id).toEqual('whatever_nihilists');
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
