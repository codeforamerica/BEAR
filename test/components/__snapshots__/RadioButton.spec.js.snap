// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should match exact snapshot when selected is false 1`] = `
<input
  className="radioButton radio-button"
  defaultChecked={false}
  id="whatever_nihilists"
  name="nihilists"
  onChange={[Function]}
  type="radio"
/>
`;

exports[`RadioButton component should match exact snapshot when selected is true 1`] = `
<input
  className="radioButton radio-button"
  defaultChecked={true}
  id="whatever_nihilists"
  name="nihilists"
  onChange={[Function]}
  type="radio"
/>
`;
